SonarQube.Client/Requests/PagedRequestBase.cs

```diff
@@ -60,13 +65,24 @@ public async override Task<TResponseItem[]> InvokeAsync(HttpClient httpClient, C
 
                 Page++;
             }
-            while (pageResult.Value != null &&
-                // Continue paging until we get a partial page of results i.e. fewer than requested.
-                // NB there is a bug here: should be comparing against the request page size, not the
-                // maximum allowed size. See https://github.com/SonarSource/sonarqube-webclient-dotnet/issues/8
-                pageResult.Value.Length >= MaximumPageSize);
+            while (!ReachedMaxPageNumber() &&
+                pageResult.Value != null &&
+                pageResult.Value.Length >= PageSize);
+
+            return allResponseItems.Take(ItemsLimit).ToArray();
```

Comment:
This will be redundant in 99.99% of cases, and it's not immediately obvious why it is there.
I think it would be clearer and more obvious if this were only done if there are more items than requested.